packages/files-ui/src/Components/Modules/Settings/SubscriptionPlan/AddCardModal.tsx

```diff
@@ -0,0 +1,237 @@
+          borderTopLeftRadius: `${constants.generalUnit * 1.5}px`,
+          borderTopRightRadius: `${constants.generalUnit * 1.5}px`,
+          maxWidth: `${breakpoints.width("md")}px !important`
+        }
+      },
+      okButton: {
+        marginLeft: constants.generalUnit
+      },
+      cancelButton: {
+        [breakpoints.down("md")]: {
+          position: "fixed",
+          bottom: 0,
+          left: 0,
+          width: "100%",
+          height: constants?.mobileButtonHeight
+        }
+      },
+      label: {
+        fontSize: 14,
+        lineHeight: "22px"
+      },
+      heading: {
+        color: constants.createFolder.color,
+        fontWeight: typography.fontWeight.semibold,
+        textAlign: "left",
+        marginBottom: constants.generalUnit * 2
+      },
+      footer: {
+        marginTop: constants.generalUnit * 4
+      }
+    })
+  }
+)
+
+interface ICreateFolderModalProps {
+  isModalOpen: boolean
+  onClose: () => void
+}
+
+const CreateFolderModal: React.FC<ICreateFolderModalProps> = ({
+  isModalOpen,
+  onClose
+}: ICreateFolderModalProps) => {
+  const classes = useStyles()
+  const [cardInputs, setCardInputs] = useState({
+    cardNumber: "",
+    cardExpiry: "",
+    cardCvc: ""
+  })
+  const [cardName, setCardName] = useState("")
+  const [error, setError] = useState("")
+  const [loading, setLoading] = useState(false)
+  const { filesApiClient } = useFilesApi()
+  const { addToastMessage } = useToaster()
+
+  const onCloseModal = () => {
+    setCardInputs({ cardNumber: "", cardExpiry: "", cardCvc: "" })
+    setCardName("")
+    setError("")
+    onClose()
+  }
+
+  const onSubmitCard = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    const error = !cardName
+      ? t`Name on card is required`
+      : getCardNumberError(cardInputs.cardNumber) ||
+      getExpiryDateError(cardInputs.cardExpiry) ||
+      getCVCError(cardInputs.cardCvc, cardInputs.cardNumber)
+
+    if (error) {
+      setError(error)
+      return
+    }
+    setError("")
+
+    setLoading(true)
+    // get token from stripe
+    axios({
+      method: "post",
+      url: "https://api.stripe.com/v1/tokens",
+      headers: {
+        "Authorization": `Bearer ${process.env.REACT_APP_STRIPE_PK}`,
+        "Content-Type": "application/x-www-form-urlencoded"
+      },
+      data : qs.stringify({
+        "card[number]": cardInputs.cardNumber,
+        "card[exp_month]": cardInputs.cardExpiry.split("/")[0].trim(),
+        "card[exp_year]": cardInputs.cardExpiry.split("/")[1].trim(),
+        "card[cvc]": cardInputs.cardCvc
+      })
+    }).then((resp: AxiosResponse<{ card: {id: string}}>) => {
+      // send stripe token to API
+      filesApiClient.addCard({ token: resp.data.card.id })
+        .then(() => {
+          onCloseModal()
+          addToastMessage({
+            message: t`Card added successfully`
+          })
+        }).catch(() => {
+          setError(t`Something went wrong`)
```

Comment:
We should be a little more helpful to our users and ask to try again maybe?
Also to help us, logging in the console could help
```suggestion
        }).catch((e) => {
          setError(t`Something went wrong, please try again`)
          console.error(e)
```